External/SPMmodified/spm_load.m

```diff
@@ -191,3 +225,50 @@
     x = struct2cell(x);
     x = [x{:}];
 end
+
+
+end
+
+%% ========================================================================
+
+function [d] = FixIllegalEmptyCells(d, N)
+%FixIllegalEmptyCells % EXPLOREASL HACK: Fix potentially erroneous empty cells
+%   d = all cells from input delimiter-separate value file (without header)
+%   N = number of columns
+    
+    illegalEmptyCells = DetectIllegalEmptyCells(d, N);
+
+    while ~isempty(illegalEmptyCells)
+        illegalRow = illegalEmptyCells(1);
+        fprintf('%s\n', ['Found empty cells at row ' xASL_num2str(illegalRow)]);
+
+        % Remove all adjacent empty cells
+        while isempty(d{1}{illegalRow})
+            d{1}(illegalRow:end-1) = d{1}(illegalRow+1:end);
+            d{1} = d{1}(1:end-1);
+        end
+
+        illegalEmptyCells = DetectIllegalEmptyCells(d, N);
+    end
+
+end
+
+%% ========================================================================
+
+
+function [illegalEmptyCells] = DetectIllegalEmptyCells(d, N)
+%DetectIllegalEmptyCells % EXPLOREASL HACK: detect & fix potentially erroneous empty cells
+%   d = all cells from input delimiter-separate value file (without header)
+%   N = number of columns
+    
+    emptyCells = find(cellfun(@(y) isempty(y), d{1}));
+    % N = number of columns
+    % so if any index-1 of the empty cells equals to the number of columns,
+    % this means that the empty cell is at the end of a line and potentially
+    % crashing this function trying to open a delimiter-separated value file
+    % (dsv)
+    
+    illegalEmptyCells = (emptyCells-1)/N == round((emptyCells-1)/N);
```

Comment:
That will not work correctly if N is an integer. Correct function is "mod(emptyCells-1,N) == 0"

Reply:
Besides - what happens if there's an entire empty row, then this will be fine. What happens if half of a row is empty and half of another row is empty?